### ISSUE TITLE
local dockerless managed VMs should not force URLFetch

### ISSUE THREAD

**Issue by GoogleCodeExporter** [prototipando/google-cloud-sdk#254]
```
Otherwise the logs are full of these when you use urllib(2). 

Stripped prohibited headers from URLFetch request: ['Content-Length', 'Host']
```

Original issue reported on code.google.com by `pdknsk` on 16 Sep 2015 at 8:30

**Comment by GoogleCodeExporter** [comment]
```
Just to expand: in App Engine every request by urllib(2) is routed through 
URLFetch, which has the above mentioned restriction. Except that it doesn't 
apply to Managed VMs, in which case routing through URLFetch should be disabled 
in the development server when using local dockerless Managed VMs.
```

Original comment by `pdknsk` on 17 Sep 2015 at 4:58

**Comment by GoogleCodeExporter** [comment]
Original comment by `dlor...@google.com` on 22 Sep 2015 at 4:32

**Comment by GoogleCodeExporter** [comment]
Original comment by `gsfow...@google.com` on 23 Sep 2015 at 1:37
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the comment. We are looking into it.
```

Original comment by `gsfow...@google.com` on 23 Sep 2015 at 7:19

**Comment by GoogleCodeExporter** [comment]
```
I've noticed a few other problems caused by this, like exception handling.
```

Original comment by `pdknsk` on 24 Sep 2015 at 3:18

**Comment by GoogleCodeExporter** [comment]
```
culprit: 
google-cloud-sdk/platform/google_appengine/google/appengine/dist27/httplib.py
```

Original comment by `pdknsk` on 24 Sep 2015 at 6:33

**Comment by GoogleCodeExporter** [comment]
```
There is at least one additional layer below, namely socket reimplemented.
```

Original comment by `pdknsk` on 24 Sep 2015 at 9:47

**Comment by GoogleCodeExporter** [comment]
```
Slight correction: while socket is reimplemented, that doesn't seem to
cause any problems or unexpected results. The exception problem I had
was my fault. It should be sufficient to just disable the httplib
override.
```

Original comment by `pdknsk` on 26 Sep 2015 at 7:25

**Comment by GoogleCodeExporter** [comment]
```
This is fixed as of 0.9.85 – perhaps en passant, because I didn't notice a 
message about it in the release notes.
```

Original comment by `pdknsk` on 11 Nov 2015 at 9:32

**Comment by GoogleCodeExporter** [comment]
```
Thanks for verifying.
```

Original comment by `dlor...@google.com` on 11 Nov 2015 at 9:33
- Changed state: **Fixed**
